gateway/cert.go

```diff
@@ -381,8 +381,15 @@ func (gw *Gateway) getTLSConfigForClient(baseConfig *tls.Config, listenPort int)
 		defer gw.apisMu.RUnlock()
 
 		newConfig.ClientCAs = x509.NewCertPool()
-
 		domainRequireCert := map[string]tls.ClientAuthType{}
+
+		directMTLSDomainMatch := false
+		for _, spec := range gw.apiSpecs {
+			if spec.UseMutualTLSAuth && spec.Domain == hello.ServerName {
+				directMTLSDomainMatch = true
```

Comment:
we can break the loop after `directMTLSDomainMatch` is true, skipping iteration over rest of the apis

Reply:
done